mule-migration-tool-library/src/main/java/com/mulesoft/tools/migration/library/mule/steps/core/MigrateApiConfigFlows.java

```diff
@@ -0,0 +1,198 @@
+
+import com.mulesoft.tools.migration.library.apikit.steps.AbstractApikitMigrationStep;
+import com.mulesoft.tools.migration.step.category.MigrationReport;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jdom2.CDATA;
+import org.jdom2.Element;
+import org.raml.v2.api.RamlModelBuilder;
+import org.raml.v2.api.RamlModelResult;
+import org.raml.v2.api.model.v10.datamodel.TypeDeclaration;
+import org.raml.v2.api.model.v10.methods.Method;
+import org.raml.v2.api.model.v10.resources.Resource;
+
+/**
+ * Migrate ApiConfig flow definitions
+ * 
+ * @author Mulesoft Inc.
+ * @since 1.0.0
+ */
+public class MigrateApiConfigFlows extends AbstractApikitMigrationStep {
+
+
+  private static final String APIKIT_NAMESPACE_URI = "http://www.mulesoft.org/schema/mule/mule-apikit";
+  private static final String XPATH_SELECTOR_ROUTER =
+      "//*[local-name()='router' and namespace-uri()='" + APIKIT_NAMESPACE_URI + "']";
+  public static final String XPATH_SELECTOR = getTopLevelCoreXPathSelector("flow");
+  static final String MULE_4_API_FOLDER = "src" + File.separator + "main" + File.separator + "resources" + File.separator + "api";
+
+  RamlModelBuilder builder = null;
+  RamlModelResult result = null;
+
+  @Override
+  public String getDescription() {
+    return "Migrate ApiConfig flow definitions";
+  }
+
+  public MigrateApiConfigFlows() {
+    this.setAppliedTo(XPATH_SELECTOR);
+  }
+
+  @Override
+  public void execute(Element element, MigrationReport report) throws RuntimeException {
+
+    // resolve config-ref
+    String apiConfig = null;
+
+    final Element node = getApplicationModel().getNode(XPATH_SELECTOR_ROUTER);
+    if (node != null && node.getAttribute("config-ref") != null) {
+      apiConfig = node.getAttribute("config-ref").getValue();
+
+    }
+    if (apiConfig != null) {
+      // API Kit Router found
+      addNameSpace(CORE_EE_NAMESPACE, EE_NAMESPACE_SCHEMA, element.getDocument());
+
+
+      final String flowName = element.getAttributeValue("name");
+
+      if (flowName.endsWith(":" + apiConfig)) {
+
+        List<String> uriParams = getResourceByFlowName(flowName.replaceAll(":" + apiConfig, ""), report, element);
+        if (uriParams != null && uriParams.size() > 0) {
+          Element transformElement = new Element("transform", CORE_EE_NAMESPACE);
+          transformElement.removeContent();
+          final Element variablesElement = new Element("variables", CORE_EE_NAMESPACE);
+          variablesElement.removeContent();
+          transformElement.addContent(variablesElement);
+
+          for (String name : uriParams) {
+            final Element variableElement = new Element("set-variable", CORE_EE_NAMESPACE);
+            variableElement.setAttribute("variableName", name);
+            variableElement.setContent(new CDATA("attributes.uriParams." + name));
+            variablesElement.addContent(variableElement);
```

Comment:
```suggestion
            variablesElement.addContent(variableElement);
            report.report("raml.uriParams", element, variableElement, name);
```

Reply:
In a new API, the transformation for the URI parameters is created at the scaffold of the Raml. Thus, this should also be done during the migration and this without a hint. This confuses the user what is correct.